src/Simulation/TargetDefinitions/Intrinsic/Z.qs

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
```

Comment:
I might have missed something earlier, but what is the reason to put each operation in its own file?

Reply:
It's not obvious when there is only one target definition how this would be used, but you can check out a draft PR I have showing the ongoing work on the Type2 definition to see how this extends to multiple packages: #353. My intent with that draft is to close it and make a fresh one off of a cleaned up branch once this PR gets in. I'm still working on finding ways to reduce renames and duplications in order to make that PR more legible.

To describe the idea briefly, the set of intrinsics and decompositions for any and all target definitions should live in these folders. Having them in separate files then allows us to define a new target package via a list of `QsharpCompile` references to the specific subset of intrinsics and decompositions that make up that target definition (see [Type2.Package.props](https://github.com/microsoft/qsharp-runtime/pull/353/files#diff-233678157037e3e3be828fb427f29d49) as an example of this). This way we can easily reuse intrinsics or decompositions across targets, with the trade off being an explosion in the number of files. As you noted, this does make the first package look kinda silly here, but doing two at once would be way too many files and make the PR very hard to review.

Reply:
Thanks for the explanation, I think it answered my question of what happens if there are multiple decompositions for a given operation: it looks like there are multiple files that define the same operation, and targets only pick one of them.